pkg/util/pod/pod.go

```diff
@@ -56,3 +62,36 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 		return g.Name == gateName
 	})
 }
+
+func ReadUIntFromLabel(pod *corev1.Pod, labelKey string) (*int, error) {
+	return ReadUIntFromLabelWithMinMax(pod, labelKey, math.MaxInt)
+}
+
+func ReadUIntFromLabelWithMinMax(obj client.Object, labelKey string, max int) (*int, error) {
+	value, found := obj.GetLabels()[labelKey]
+	if !found {
+		return nil, fmt.Errorf("no label %q for %s %q", labelKey, obj.GetObjectKind(), klog.KObj(obj))
+	}
+	intValue, err := readUIntFromStringWithMax(value, max)
+	if err != nil {
+		return nil, fmt.Errorf("incorrect label value %q for %s %q: %w", value, obj.GetObjectKind(), klog.KObj(obj), err)
+	}
+	return intValue, nil
+}
+
+func readUIntFromStringWithMax(value string, max int) (*int, error) {
+	return readIntFromStringWithMinMax(value, 0, max)
+}
+
+func readIntFromStringWithMinMax(value string, min, max int) (*int, error) {
+	i, err := strconv.Atoi(value)
+	if err != nil {
+		return nil, err
+	}
+	if i < min {
+		return nil, fmt.Errorf("value should be greater than or equal to %d", min)
```

Comment:
Could we reduce this code and use `strconv.ParseUInt`? Since we don't export those function I believe `readIntFromStringWithMax` is enough

Reply:
Done

Reply:
Ah, missed ParseUint

Reply:
Done